prober/http_test.go

```diff
@@ -164,6 +164,33 @@ func TestPost(t *testing.T) {
 	}
 }
 
+func TestPostBody(t *testing.T) {
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if r.Method != "POST" {
+			w.WriteHeader(http.StatusBadRequest)
+		}
+	}))
+	defer ts.Close()
+
+	recorder := httptest.NewRecorder()
+	registry := prometheus.NewRegistry()
+	testCTX, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+	result := ProbeHTTP(testCTX, ts.URL,
+		config.Module{
+			Timeout: time.Second,
+			HTTP: config.HTTPProbe{
+				IPProtocolFallback: true,
+				Method:             "POST",
+				BodyFile:           "./http.go",
```

Comment:
You can create a testdata directory, and a file with known contents there - plus verify that's what's coming through.

Reply:
You should still change this to use explicit testdata files, rather than the source itself